### ISSUE TITLE
Allow to attach the process based on Process ID

### ISSUE THREAD

**Issue by karthikmsctcs** [microsoft/WinAppDriver#1091]
Appium option should allow to attach the currently running process based on the Process ID

**Comment by naeemakram** [comment]
This is a feature request.

**Comment by karthikmsctcs** [comment]
This will attach the currently running process to session without using Root session

var processes = Process.GetProcesses();
            bool exi = false;
            foreach (var process in processes)
            {
                
                string v = "";
                try
                {
                    v = process.MainModule.FileName;
                }
                catch { }
                if (v.ToUpper() == ApplicationPath.ToUpper())
                {
                    WindowHandle = process.MainWindowHandle.ToString("x");
                    exi = true;
                    break;
                }
            }

            //////////
            AppiumOptions Prop = new AppiumOptions();
            Prop.AddAdditionalCapability("appTopLevelWindow", WindowHandle);
            Prop.AddAdditionalCapability("deviceName", "WindowsPC");
            Prop.AddAdditionalCapability("platformName", "Windows");

            Session = new WindowsDriver<WindowsElement>(new Uri(@"http://localhost:4723/"), Prop);

**Comment by naeemakram** [comment]
Thanks very much for the answer, I didn't know it was a possibility.

**Comment by Shakevg** [comment]
@naeemakram you can use next method https://github.com/microsoft/WinAppDriver/issues/534#issuecomment-562030077

**Comment by naeemakram** [comment]
> @naeemakram you can use next method [#534 (comment)](https://github.com/microsoft/WinAppDriver/issues/534#issuecomment-562030077)

Thanks for the tip

**Comment by henry1999sg** [comment]
Process.GetProcesses() is working on local machine. how can we get processes in remote machine ? via remote WinAppDriver, I start an application, the application has a separate dialog. from Inspect, it is independent from the main session but has the same pid. my problem is how to get a windows handle for that dialog. windAppDriver.WindowHandles only give me the main windows. 
thanks and regards

**Comment by karthikmsctcs** [comment]
Yes it is not possible to get Window handle from a remote machine. If any one can give suggestion to get WindowsHandle from a remote machine will be very helpful.

**Comment by huster-songtao** [comment]
> Yes it is not possible to ge

Create new console app, the console app supply grpc or http service on server, response processes to clients,  
the console app start with the codes blow:
```
            var startInfo = new ProcessStartInfo("C:\\Program Files\\Windows Application Driver\\WinAppDriver.exe");
            var winAppDriverProcess = Process.Start(startInfo);
```

The client can get processes from remote machine.

**Comment by huster-songtao** [comment]
> 

The ability to minimize the application to the Taskbar is a standard behavior of the vast majority of Windows applications. we have an option to minimize (or close) the program to the system tray (officially known as the Notification Area) that is usually located next to the Taskbar in the bottom-right corner. 

process.MainWindowHandle is 0x0000000000000000
Session can not created by process.MainWindowHandle

**Comment by sajid36** [comment]
I found a workaround, find your desired applications WindowHandle, then directly create session using extracted WindowHandle. Traversing from root is not worthy. I found a lot of people are complaining about how to switch I hope this will help a lot!

```
import win32gui

def findWindowHandle():
    handleList = []
    def findit(hwnd,ctx):
        if win32gui.GetWindowText(hwnd) == "Microsoft Store": # check the title
            handleList.append(hwnd)

    win32gui.EnumWindows(findit,None)
    return handleList

handle = hex(findWindowHandle()[0])
print(handle)


desired_caps = {}
desired_caps["appTopLevelWindow"] = handle
desired_caps["deviceName"]="WindowsPC"
desired_caps["platformName"]="Windows"
session = webdriver.Remote(command_executor='http://127.0.0.1:4723', desired_capabilities= desired_caps)

```
